core/src/boot.py

```diff
@@ -39,9 +39,11 @@
                     return
                 else:
                     label = "Wrong PIN, enter again"
-        except Exception as e:
+        except PinCancelled as e:
```

Comment:
`SdProtectCancelled` is also a legitimate exception to catch here.

`OSError` can be raised in https://github.com/trezor/trezor-firmware/blob/main-exception/core/src/apps/common/sd_salt.py#L148. This can happen if writing to the SD card is not very reliable. If that happens, then I think the proper course of action is to retry, so we should also include `OSError` here.

Reply:
Good catch. Amended in d188c1fd0740f7c5342211699b1031865445362b

Reply:
Of course if the SD card is really messed up and keeps raising OSError, then the above gets the user stuck in an infinite loop, but I still think that's better than forcing the user to unplug the device.

An alternative solution is to wrap the `request_sd_salt()` call on line 23 with try/except OSError and do 3 retries before halting.

However, I am thinking that perhaps the best solution would be to handle this situation directly inside `request_sd_salt()` by showing the `_write_failed_dialog()`, allowing the user to Retry or Abort. If they abort, then we halt.

Opinions?

Reply:
Changes in boot.py approved.
I will create a separate commit to handle this situation directly inside `request_sd_salt()` by informing the user and allowing them to retry or abort.